core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java

```diff
@@ -38,14 +43,20 @@
      */
     void writeTo(StreamOutput out) throws IOException;
 
-     @FunctionalInterface
-     interface IOFunction<T, R> {
-         /**
-          * Applies this function to the given argument.
-          *
-          * @param t the function argument
-          * @return the function result
-          */
-         R apply(T t) throws IOException;
-     }
+    @Override
+    default T readFrom(StreamInput in) throws IOException {
```

Comment:
My plan is to eventually have Writeable not extends StreamableReader at all. The whole interface might go away but I think it'll live on as a part of cluster state diffs for a while because that code is more complicated to refactor.